pkg/storage/endpoint/tso.go

```diff
@@ -64,7 +66,33 @@ func (se *StorageEndpoint) LoadTimestamp(prefix string) (time.Time, error) {
 }
 
 // SaveTimestamp saves the timestamp to the storage.
-func (se *StorageEndpoint) SaveTimestamp(key string, ts time.Time) error {
-	data := typeutil.Uint64ToBytes(uint64(ts.UnixNano()))
-	return se.Save(key, string(data))
+func (se *StorageEndpoint) SaveTimestamp(prefix string, key string, ts time.Time) error {
+	return se.RunInTxn(context.Background(), func(txn kv.Txn) error {
+		prefixEnd := clientv3.GetPrefixRangeEnd(prefix)
+		keys, values, err := txn.LoadRange(prefix, prefixEnd, 0)
+		if err != nil {
+			return err
+		}
+
+		previousTS := typeutil.ZeroTime
+		for i, key := range keys {
+			key := strings.TrimSpace(key)
+			if !strings.HasSuffix(key, timestampKey) {
+				continue
+			}
+			tsWindow, err := typeutil.ParseTimestamp([]byte(values[i]))
+			if err != nil {
+				continue
+			}
+			if typeutil.SubRealTimeByWallClock(tsWindow, previousTS) > 0 {
+				previousTS = tsWindow
+			}
+		}
+		if previousTS != typeutil.ZeroTime && typeutil.SubRealTimeByWallClock(ts, previousTS) <= 0 {
+			log.Warn("save timestamp failed, the timestamp is not bigger than the previous one", zap.Time("previous", previousTS), zap.Time("current", ts))
+			return nil
+		}
+		data := typeutil.Uint64ToBytes(uint64(ts.UnixNano()))
+		return txn.Save(key, string(data))
```

Comment:
Because LoadTimestamp() use the logic to do prefix scan then return the largest timestamp among all timestamps, it seems that we don't need the whole change in this pr -- tso service doesn't need to use the same path as pd's but the same prefix for timestamp range scan, then anyway  LoadTimestamp()  will return the largest one among the last writes of pd and tso service.